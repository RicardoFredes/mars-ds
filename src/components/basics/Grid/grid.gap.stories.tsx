import type { ComponentMeta, ComponentStory } from "@storybook/react";

import Grid from "./grid.component";
import { getGridStory } from "./grid.helpers";

export default {
  title: "Components/Grid",
  parameters: {
    docs: {
      description: {
        component:
          "Grid Responsivo. Grid é uma malha formada pela interseção de um conjunto de linhas horizontais e um conjunto de linhas verticais – um dos conjuntos define colunas e outro linhas. O parâmetro `gap` pode receber um `number` ou `text`. Nesse exemplo, passamos um `number`, sendo assim, o espaçamento entre linhas e colunas será maior ou menor a medida que esse valor aumente ou diminua. Dessa forma é possível criar um layout adaptativo aos diferentes tamanhos de telas, respeitando a regra: `mobile first`. Em outras palavras, o grid monta o layout da menor tela para a maior: `xs`, `sm`, `md`, `lg`, `xl`. Na falta de algum valor, o `columns` assume o valor atribuído à uma dimensão de tela menor, sendo 1 o valor padrão.",
      },
    },
  },
  component: Grid,
  argTypes: {
    as: {
      control: {
        type: "text",
      },
      defaultValue: "div",
      description: "Permite trocar a tag HTML, por exemplo: `section` ao invés de `div`",
    },
    columns: {
      control: {
        type: "number",
      },
      description: "Número de colunas que distribui o conteúdo na página",
      defaultValue: 2,
    },
    gap: {
      control: {
        type: "number",
      },
      description: "Espaçamento entre linhas e colunas",
      defaultValue: 24,
    },
    growing: {
      control: {
        type: "boolean",
      },
      description: "Opção que nivela o tamanho das colunas independentemente do seu conteúdo",
      defaultValue: true,
    },
  },
} as ComponentMeta<typeof Grid>;

const Template: ComponentStory<typeof Grid> = (args) => {
  return getGridStory(args);
};

const DefaultArgs = {
  as: "div",
  columns: 2,
  gap: 24,
  growing: true,
};

export const GapExample = Template.bind({});
GapExample.args = DefaultArgs;
