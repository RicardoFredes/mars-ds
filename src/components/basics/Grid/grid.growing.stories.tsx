import type { ComponentMeta, ComponentStory } from "@storybook/react";

import Card, { CardElevations } from "../Card";
import Grid from "./grid.component";

export default {
  title: "Components/Grid",
  parameters: {
    docs: {
      description: {
        component:
          "Grid Responsivo. Grid é uma malha formada pela interseção de um conjunto de linhas horizontais e um conjunto de linhas verticais – um dos conjuntos define colunas e outro linhas. O parâmetro `growing` recebe um valor booleano de `true` ou `false`. Nesse exemplo, passamos `false`, sendo assim, caso o conteúdo de uma das colunas seja maior que o outro, essa coluna será maior quanto o parâmetro `growing` está `false`, caso `true` as colunas terão o mesmo tamanho. Dessa forma é possível criar um layout adaptativo aos diferentes tamanhos de telas, respeitando a regra: `mobile first`. Em outras palavras, o grid monta o layout da menor tela para a maior: `xs`, `sm`, `md`, `lg`, `xl`. Na falta de algum valor, o `columns` assume o valor atribuído à uma dimensão de tela menor, sendo 1 o valor padrão.",
      },
      source: {
        code: `{
  "component": "Grid",
  "gap": 10, 
  "columns": 2,
  "growing": false,
  "children": [
    {
      "component": "Card",
      "elevation": "lw",
      "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of de Finibus Bonorum et Malorum (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of thics, very popular during the Renaissance. The first line of Lorem Ipsum, Lorem ipsum dolor sit amet.., comes from a line in section 1.10.32."
    },
    {
      "component": "Card",
      "elevation": "lw",
      "children": "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for lorem ipsum will uncover many web sites still in their infancy.Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like)."
    }
  ]
}`,
        language: "json",
        type: "code",
      },
    },
  },
  component: Grid,
  argTypes: {
    as: {
      control: {
        type: "text",
      },
      defaultValue: "div",
      description: "Permite trocar a tag HTML, por exemplo: `section` ao invés de `div`",
    },
    columns: {
      control: {
        type: "number",
      },
      description: "Número de colunas que distribui o conteúdo na página",
      defaultValue: 2,
    },
    gap: {
      control: {
        type: "number",
      },
      description: "Espaçamento entre linhas e colunas",
      defaultValue: 24,
    },
    growing: {
      control: {
        type: "boolean",
      },
      description: "Opção que nivela o tamanho das colunas independentemente do seu conteúdo",
      defaultValue: false,
    },
    reversedLastElement: {
      options: ["xs", "sm"],
      control: { type: "select" },
      description: "Opção que reverte o posicionamento do último elemento em mobile.",
    },
  },
} as ComponentMeta<typeof Grid>;

const Template: ComponentStory<typeof Grid> = (args) => (
  <Grid {...args}>
    <Card elevation={CardElevations.Low}>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has
      been the industrys standard dummy text ever since the 1500s, when an unknown printer took a
      galley of type and scrambled it to make a type specimen book. It has survived not only five
      centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It
      was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum
      passages, and more recently with desktop publishing software like Aldus PageMaker including
      versions of Lorem Ipsum. Contrary to popular belief, Lorem Ipsum is not simply random text. It
      has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old.
      Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of
      the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the
      cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum
      comes from sections 1.10.32 and 1.10.33 of de Finibus Bonorum et Malorum (The Extremes of Good
      and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very
      popular during the Renaissance. The first line of Lorem Ipsum, Lorem ipsum dolor sit amet..,
      comes from a line in section 1.10.32.
    </Card>
    <Card elevation={CardElevations.Low}>
      It is a long established fact that a reader will be distracted by the readable content of a
      page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less
      normal distribution of letters, as opposed to using Content here, content here, making it look
      like readable English. Many desktop publishing packages and web page editors now use Lorem
      Ipsum as their default model text, and a search for lorem ipsum will uncover many web sites
      still in their infancy. Various versions have evolved over the years, sometimes by accident,
      sometimes on purpose (injected humour and the like).
    </Card>
  </Grid>
);

const DefaultArgs = {
  as: "div",
  columns: 2,
  gap: 10,
  growing: false,
};

export const GrowingExample = Template.bind({});
GrowingExample.args = DefaultArgs;
