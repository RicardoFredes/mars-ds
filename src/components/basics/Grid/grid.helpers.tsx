import type { GridProps } from "./grid.types";

import { makeArray } from "@/services";

import Card, { CardElevations } from "../Card";
import Grid from "./grid.component";

export const getGridStory = (props: GridProps) => {
  return (
    <Grid {...props}>
      {makeArray(12).map((_item, key) => {
        return (
          <Card key={key} elevation={CardElevations.Low}>
            Column {key + 1}
          </Card>
        );
      })}
    </Grid>
  );
};
