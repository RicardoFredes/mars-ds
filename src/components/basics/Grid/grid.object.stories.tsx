import type { ComponentMeta, ComponentStory } from "@storybook/react";

import Grid from "./grid.component";
import { getGridStory } from "./grid.helpers";

export default {
  title: "Components/Grid",
  component: Grid,
  parameters: {
    docs: {
      description: {
        component:
          "Grid Responsivo. Grid é uma malha formada pela interseção de um conjunto de linhas horizontais e um conjunto de linhas verticais – um dos conjuntos define colunas e outro linhas. O parâmetro `columns` pode receber diferentes formatos de valores. Nesse exemplo, passamos um `object`, sendo assim, o layout deverá manter duas colunas para o tamanho de tela média (a partir de 768px). Dessa forma é possível criar um layout adaptativo aos diferentes tamanhos de telas, respeitando a regra: `mobile first`. Em outras palavras, o grid monta o layout da menor tela para a maior: `xs`, `sm`, `md`, `lg`, `xl`. Na falta de algum valor, o `columns` assume o valor atribuído à uma dimensão de tela menor, sendo 1 o valor padrão.",
      },
    },
  },
  argTypes: {
    as: {
      control: {
        type: "text",
      },
      defaultValue: "div",
      description: "Permite trocar a tag HTML, por exemplo: `section` ao invés de `div`",
    },
    columns: {
      control: {
        type: "object",
      },
      description:
        "Número de colunas que distribui o conteúdo na página em determinados tipos de telas (`xs`, `sm`, `md` e `lg`)",
      defaultValue: { md: 2 },
    },
    gap: {
      control: {
        type: "number",
      },
      description: "Espaçamento entre linhas e colunas",
      defaultValue: 10,
    },
    growing: {
      control: {
        type: "boolean",
      },
      description: "Opção que nivela o tamanho das colunas independentemente do seu conteúdo",
      defaultValue: true,
    },
  },
} as ComponentMeta<typeof Grid>;

const Template: ComponentStory<typeof Grid> = (args) => {
  return getGridStory(args);
};

const DefaultArgs = {
  as: "div",
  columns: { md: 2 },
  gap: 10,
  growing: true,
};

export const ObjectColumns = Template.bind({});
ObjectColumns.args = DefaultArgs;
