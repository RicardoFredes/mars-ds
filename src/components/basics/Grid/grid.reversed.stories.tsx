import type { ComponentMeta, ComponentStory } from "@storybook/react";
import Card, { CardElevations } from "../Card";

import Grid from "./grid.component";
import { getGridStory } from "./grid.helpers";

export default {
  title: "Components/Grid",
  component: Grid,
  parameters: {
    docs: {
      description: {
        component:
          "Grid Responsivo. Grid é uma malha formada pela interseção de um conjunto de linhas horizontais e um conjunto de linhas verticais – um dos conjuntos define colunas e outro linhas. O parâmetro `columns` pode receber diferentes formatos de valores. Nesse exemplo, passamos um `array`, sendo assim, o layout deverá manter três colunas - duas frações | quatro frações | 200px - independentemente do tamanho de tela. Dessa forma é possível criar um layout adaptativo aos diferentes tamanhos de telas, respeitando a regra: `mobile first`. Em outras palavras, o grid monta o layout da menor tela para a maior: `xs`, `sm`, `md`, `lg`, `xl`. Na falta de algum valor, o `columns` assume o valor atribuído à uma dimensão de tela menor, sendo 1 o valor padrão.",
      },
      source: {
        code: `{
  "component": "Grid",
  "columns": {
    "md": 2
  }, 
  "gap": 10, 
  "growing": true, 
  "reserverdLastElement": "xs", 
  "children": [
    {
      "component": "Card",
      "elevation": "lw",
      "children": "Column 1"
    },
    {
      "component": "Card",
      "elevation": "lw",
      "children": "Column 2"
    }
  ]
}`,
        language: "json",
        type: "code",
      },
    },
  },
  argTypes: {
    as: {
      control: {
        type: "text",
      },
      defaultValue: "div",
      description: "Permite trocar a tag HTML, por exemplo: `section` ao invés de `div`",
    },
    columns: {
      control: {
        type: "array",
      },
      defaultValue: { md: 2 },
      description: "Número de colunas que distribui o conteúdo na página em ordem",
    },
    gap: {
      control: {
        type: "number",
      },
      description: "Espaçamento entre linhas e colunas",
      defaultValue: 10,
    },
    growing: {
      control: {
        type: "boolean",
      },
      defaultValue: true,
      description: "Opção que nivela o tamanho das colunas independentemente do seu conteúdo",
    },
    reversedLastElement: {
      options: ["xs", "sm"],
      defaultValue: "xs",
      control: { type: "select" },
      description: "Opção que reverte o posicionamento do último elemento em mobile.",
    },
  },
} as ComponentMeta<typeof Grid>;

const Template: ComponentStory<typeof Grid> = (args) => (
  <Grid {...args}>
    <Card elevation={CardElevations.Low}>Column 1</Card>
    <Card elevation={CardElevations.Low}>Column 2</Card>
  </Grid>
);

const DefaultArgs = {
  as: "div",
  columns: { md: 2 },
  gap: 10,
  growing: true,
};

export const ReversedColumns = Template.bind({});
ReversedColumns.args = DefaultArgs;
