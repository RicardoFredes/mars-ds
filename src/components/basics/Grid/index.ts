/**
 * WARNING!
 * Do not change this file,
 * because it is automatically generated
 * run "yarn reindex" to update
 */

import Grid from "./grid.component";
export default Grid;
export * from "./grid.types";
