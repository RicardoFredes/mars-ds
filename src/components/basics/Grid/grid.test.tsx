import type { GridProps } from "./grid.types";

import { render } from "@testing-library/react";

import Grid from "./grid.component";

const makeSut = (props?: GridProps) => render(<Grid {...props} />);

describe("<Grid>", () => {
  it("should render component", () => {
    const wrapper = makeSut();
    expect(wrapper).toMatchSnapshot();
  });

  describe("grid with number columns", () => {
    it("render with number columns", () => {
      const wrapper = makeSut({
        columns: 2,
      });
      expect(wrapper).toMatchSnapshot();
    });
    it("verify style into grid", () => {
      const wrapper = makeSut({
        columns: 2,
      });
      const element = wrapper.getByTestId("grid");
      expect(element).toHaveAttribute(
        "style",
        "--columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
      );
    });
    it("verify classnames into grid", () => {
      const wrapper = makeSut({
        columns: 2,
        growing: false,
      });
      const element = wrapper.getByTestId("grid");
      expect(element).toHaveAttribute("class", "grid");
    });
    it("verify classnames gap into grid", () => {
      const wrapper = makeSut({
        columns: 3,
        gap: 10,
      });
      const element = wrapper.getByTestId("grid");
      expect(element).toHaveAttribute(
        "style",
        "--gap: 10px; --columns-xs: repeat(3, 1fr); --columns-sm: repeat(3, 1fr); --columns-md: repeat(3, 1fr); --columns-lg: repeat(3, 1fr);"
      );
    });
    it("render with gap property", () => {
      const wrapper = makeSut({
        columns: 2,
        gap: 10,
      });
      expect(wrapper).toMatchSnapshot();
    });
    it("verify classnames growing into grid", () => {
      const wrapper = makeSut({
        columns: 2,
        growing: true,
      });
      const element = wrapper.getByTestId("grid");
      expect(element).toHaveAttribute("class", "grid grid--growing");
    });
    it("render with growing property", () => {
      const wrapper = makeSut({
        columns: 2,
        gap: 10,
        growing: false,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("grid with string columns", () => {
    it("render with string columns", () => {
      const wrapper = makeSut({
        columns: "200px 1fr",
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("grid with object columns", () => {
    it("render with object columns", () => {
      const wrapper = makeSut({
        columns: { md: 2, lg: 3 },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("grid with array columns", () => {
    it("render with array columns", () => {
      const wrapper = makeSut({
        columns: [1, 2],
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
