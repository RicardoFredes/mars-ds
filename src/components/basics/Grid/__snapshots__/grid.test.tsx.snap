// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid> grid with array columns render with array columns 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid--growing"
        data-testid="grid"
        style="--columns-xs: 1fr 2fr; --columns-sm: 1fr 2fr; --columns-md: 1fr 2fr; --columns-lg: 1fr 2fr;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid--growing"
      data-testid="grid"
      style="--columns-xs: 1fr 2fr; --columns-sm: 1fr 2fr; --columns-md: 1fr 2fr; --columns-lg: 1fr 2fr;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Grid> grid with number columns render with gap property 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid--growing"
        data-testid="grid"
        style="--gap: 10px; --columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid--growing"
      data-testid="grid"
      style="--gap: 10px; --columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Grid> grid with number columns render with growing property 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid"
        data-testid="grid"
        style="--gap: 10px; --columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid"
      data-testid="grid"
      style="--gap: 10px; --columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Grid> grid with number columns render with number columns 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid--growing"
        data-testid="grid"
        style="--columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid--growing"
      data-testid="grid"
      style="--columns-xs: repeat(2, 1fr); --columns-sm: repeat(2, 1fr); --columns-md: repeat(2, 1fr); --columns-lg: repeat(2, 1fr);"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Grid> grid with object columns render with object columns 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid--growing"
        data-testid="grid"
        style="--columns-xs: 1fr; --columns-sm: 1fr; --columns-md: repeat(2, 1fr); --columns-lg: repeat(3, 1fr);"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid--growing"
      data-testid="grid"
      style="--columns-xs: 1fr; --columns-sm: 1fr; --columns-md: repeat(2, 1fr); --columns-lg: repeat(3, 1fr);"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Grid> grid with string columns render with string columns 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid--growing"
        data-testid="grid"
        style="--columns-xs: 200px 1fr; --columns-sm: 200px 1fr; --columns-md: 200px 1fr; --columns-lg: 200px 1fr;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid--growing"
      data-testid="grid"
      style="--columns-xs: 200px 1fr; --columns-sm: 200px 1fr; --columns-md: 200px 1fr; --columns-lg: 200px 1fr;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Grid> should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid--growing"
        data-testid="grid"
        style="--columns-xs: 1fr; --columns-sm: 1fr; --columns-md: 1fr; --columns-lg: 1fr;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid--growing"
      data-testid="grid"
      style="--columns-xs: 1fr; --columns-sm: 1fr; --columns-md: 1fr; --columns-lg: 1fr;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
